### PR TITLE
Add viewing beacon uses panel

### PR DESCRIPTION
**Pull request by matthew-a-carr**
## Context

- Adds the view for beacon uses
- Communication fields should only be displayed if the user has selected them within the form - see the wireframes for an example on this
- I have left out the spaces after a call with Konrad.  He said that the lines within the entries should make it easier for users to determine what field they are looking at and I suggested further down the line we might want to add sub-headings to split out the information, i.e: Aircraft summary, Aircraft Comms etc...

## Changes in this pull request

- Adds a Maritime, Aviation, and Land summary view which takes in an `IUse` and correctly renders the view
- I haven't tried to remove duplication in the views as I felt it would damage the readability of each component and could make it more difficult to manage updating in the future due to the coupling between display names of fields

## Guidance to review

- Apologies for it being a fairly large PR, on reflection I could have split out the `UsesGateway` from the views that have been created
- The majority of the code is adding of new views for each of the different beacon use types

## Link to Trello card

- [Link to Miro](https://miro.com/app/board/o9J_lZuM9qs=/?moveToWidget=3074457356299737554&cot=14)
- [Link to Trello](https://trello.com/c/U5rbX1tU/580-beacon-use-tab-readonly)

Example view: 
image

## Things to check

- [ ] Environment variables have been updated

### CODE REVIEW

File: src/entities/IUse.ts [hunk 1]
```diff
@@ -61,4 +61,8 @@ export enum Activities {
   FishingVessel = "FISHING_VESSEL",
   ClimbingMountaineering = "CLIMBING_MOUNTAINEERING",
   Glider = "GLIDER",
+  SailingVessel = "SAILING_VESSEL",
+  Other = "OTHER",
+  WorkingRemotely = "WORKING_REMOTELY",
+  Windfarm = "WINDFARM",
```
Comment (matthew-a-carr) [inline src/entities/IUse.ts:67]: I have only added in the additional activities required for the `Land` view and those used in tests

File: src/fixtures/singleBeaconApiResponse.fixture.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ const getUseResponseJson = (mainUse: boolean) => ({
   ssrNumber: "2",
   rssNumber: "3",
   officialNumber: "4",
-  rigPlatformLocation: "5",
+  rigPlatformLocation: "Scotland",
```
Comment (matthew-a-carr) [inline src/fixtures/singleBeaconApiResponse.fixture.ts:32]: Better representative data

File: src/panels/usesSummaryPanel/UseSummaryPanel.test.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+describe("UseSummaryPanel", () => {
+  let use: any;
+
+  beforeEach(() => {
+    use = {
+      activity: Activities.SailingVessel,
+      purpose: Purposes.Pleasure,
+    };
+  });
```
Comment (zackads) [inline src/panels/usesSummaryPanel/UseSummaryPanel.test.tsx:14]: This `use` as it appears in the `beforeEach` hook is only used in the first test, with the other two updating it for different uses.  Could this be in-lined to each test to improve readability?

File: src/panels/usesSummaryPanel/UseSummaryPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  use: IUse;
+  titlePrefix: string;
+}
+
+export const UseSummaryPanel: FunctionComponent<UseSummaryPanelProps> = ({
+  use,
+  titlePrefix,
+}: UseSummaryPanelProps): JSX.Element => {
+  const title = getCardHeaderTitle(titlePrefix, use);
+  const useSummary = getUseSummary(use);
```
Comment (zackads) [inline src/panels/usesSummaryPanel/UseSummaryPanel.tsx:18]: Is prefixing every function here with `get` necessary?  `cardHeaderTitle()` and `useSummary()` work fine for me.  This is very Javaland-y 😝

File: src/panels/usesSummaryPanel/UseSummaryPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    use.activity === Activities.Other
+      ? use.otherActivity?.toUpperCase()
+      : use.activity.replace(/_/, " ");
+
+  if (use.purpose) {
+    title += ` (${use.purpose})`;
+  }
+
+  return title;
+};
```
Comment (zackads) [inline src/panels/usesSummaryPanel/UseSummaryPanel.tsx:42]: This looks _very similar_ if not identical to a composite of `formatUse` and `titleCase` in `mcaWritingStyleFormatter`.  Could those functions be re-used instead?

File: src/panels/usesSummaryPanel/UseSummaryPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+const getMaritimeSummary = (use: IUse): ReactNode => (
+  <MaritimeSummary use={use} />
+);
+
+const getAviationSummary = (use: IUse): ReactNode => (
+  <AviationSummary use={use} />
+);
+
+const getLandSummary = (use: IUse): ReactNode => <LandSummary use={use} />;
```
Comment (zackads) [inline src/panels/usesSummaryPanel/UseSummaryPanel.tsx:65]: What is the value of extracting the `getter` functions?  They're only used here.  Why not

```suggestion
const getUseSummary = (use: IUse): ReactNode => {
  switch (use.environment) {
    case Environments.Maritime:
        return <MaritimeSummary use={use} />
    case Environments.Aviation:
        return <AviationSummary use={use} />
    case Environments.Land:
        return <LandSummary use={use} />
    default:
        throw Error("Unrecognized environment")
  }
};
```

File: src/panels/usesSummaryPanel/UsesListSummaryPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        console.error(error);
+      }
+    };
+
+    fetchUses(beaconId);
+
+    return () => {
+      isMounted = false;
+    };
+  }, [beaconId, usesGateway]);
```
Comment (zackads) [inline src/panels/usesSummaryPanel/UsesListSummaryPanel.tsx:36]: What's the purpose of `isMounted`? The component works fine without it.  Testing it locally and so far as I can tell the behaviour is identical (API calls, rendering etc.) both with and without that code?

**Review by zackads — CHANGES_REQUESTED** [review]
Nice work!  I haven't explicitly said "can we change this" in some of the comments, but I've asked some questions that would be good to understand your thinking before merging if that's OK.  Happy to do so on a call 👍🏻

File: src/gateways/UsesGateway.test.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    gateway = new UsesGateway(beaconResponseMapper);
+    beaconId = "f48e8212-2e10-4154-95c7-bdfd061bcfd2";
+  });
+
+  describe("fetching uses for a given beacon id", () => {
+    it("returns the uses array", async () => {
+      // @ts-ignore
+      axios.get.mockImplementationOnce(() => Promise.resolve({ data: {} }));
+      // @ts-ignore
+      beaconResponseMapper.map.mockReturnValue({ uses: usesFixture });
```
[the remark this answers is not in the document]
Comment (matthew-a-carr) [reply src/gateways/UsesGateway.test.ts:29]: Done

File: src/gateways/UsesGateway.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import axios from "axios";
+import { applicationConfig } from "config";
+import { IUse } from "entities/IUse";
+import { IUsesGateway } from "./IUsesGateway";
+import { IBeaconResponseMapper } from "./mappers/BeaconResponseMapper";
+
+export class UsesGateway implements IUsesGateway {
+  private _beaconResponseMapper;
```
[the remark this answers is not in the document]
Comment (matthew-a-carr) [reply src/gateways/UsesGateway.ts:8]: Done

File: src/panels/usesSummaryPanel/AviationSummary.test.tsx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    expect(await screen.findByText("M-ZYXW")).toBeVisible();
+    expect(await screen.findByText("YES")).toBeVisible();
+    expect(
+      await screen.findByText("STOWED INSIDE THE NOSE OF THE AIRCRAFT")
+    ).toBeVisible();
+    expect(await screen.findByText("IN MY CARRY BAG")).toBeVisible();
+  });
+
+  it("should display NO if the beacon is not a dongle", async () => {
+    const use: any = {
```
[the remark this answers is not in the document]
Comment (matthew-a-carr) [reply src/panels/usesSummaryPanel/AviationSummary.test.tsx:37]: Will typecast to `as IUse` as discussed

File: src/panels/usesSummaryPanel/AviationSummary.test.tsx [hunk 11]
```diff
@@ -0,0 +1,85 @@
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import { Placeholders } from "useCases/mcaWritingStyleFormatter";
+import { AviationSummary } from "./AviationSummary";
+
+describe("Aviation Summary", () => {
+  it("should display the aircraft summary", async () => {
+    const use: any = {
```
[the remark this answers is not in the document]
Comment (matthew-a-carr) [reply src/panels/usesSummaryPanel/AviationSummary.test.tsx:8]: Same as below, have type case to `IUse`

File: src/panels/usesSummaryPanel/UseSummaryPanel.test.tsx (hunk 3 above)
Comment (matthew-a-carr) [reply src/panels/usesSummaryPanel/UseSummaryPanel.test.tsx]: Have updated to remove the `beforeEach`

File: src/panels/usesSummaryPanel/UseSummaryPanel.tsx (hunk 5 above)
Comment (matthew-a-carr) [reply src/panels/usesSummaryPanel/UseSummaryPanel.tsx]: Used `formatUse` function

**Pull request merged**